lib/redis/index.ts

```diff
@@ -277,7 +278,10 @@ Redis.prototype.connect = function (callback) {
 
     this.condition = {
       select: options.db,
-      auth: options.password,
+      auth:
+        options.username && options.username.length > 0
```

Comment:
```suggestion
        options.username
```

We don't need to check `.length` to detect empty strings right?

Reply:
Same reasoning as above, and agree with the change.